init.lua

```diff
@@ -469,47 +469,47 @@ require('lazy').setup({
           -- to define small helper and utility functions so you don't have to repeat yourself.
           --
           -- In this case, we create a function that lets us more easily define mappings specific
-          -- for LSP related items. It sets the mode, buffer and description for us each time.
-          local map = function(keys, func, desc)
-            vim.keymap.set('n', keys, func, { buffer = event.buf, desc = 'LSP: ' .. desc })
+          -- for LSP related items. It sets the buffer and description for us each time.
+          local map = function(mode, keys, func, desc)
+            vim.keymap.set(mode, keys, func, { buffer = event.buf, desc = 'LSP: ' .. desc })
```

Comment:
you could do this:
```lua
          local map = function(keys, func, desc, mode)
            mode = mode or 'n'
            vim.keymap.set(mode, keys, func, { buffer = event.buf, desc = 'LSP: ' .. desc })
          end

```
and set the mode `{ n, x }` for that one line without changing the rest:
```lua
          map('<leader>ca', vim.lsp.buf.code_action, '[C]ode [A]ction', { 'n', 'x' })

```

Reply:
Hey @bayramkzk what do you think of this suggestion?

Reply:
I think having mode at the end of arguments seems a bit weird but I agree that this commit changed more lines than it should and as this is a very small change in the config, I don't think there is any need to debate any more about it. Reverting and recommitting with minimal changes now!